src/aiida/orm/nodes/data/remote/stash/base.py

```diff
@@ -39,13 +39,16 @@ class RemoteStashData(Data):
     class Model(Data.Model):
         stash_mode: StashMode = MetadataField(description='The mode with which the data was stashed')
 
-    def __init__(self, stash_mode: StashMode, **kwargs):
+    def __init__(self, stash_mode: StashMode, source_uuid: str, **kwargs):
         """Construct a new instance
 
         :param stash_mode: the stashing mode with which the data was stashed on the remote.
+        :param source_uuid: the uuid of the source node (str). If the source node itself is a stashed node,
+            then the `source_uuid` of the stashed node should be passed.
```

Comment:
Improve the docstring, not so clear what the source node can be besides a stashed node.